src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -456,19 +460,33 @@ export const _handleFormAuthRedirect: ControllerHandler<
         }
         case FormAuthType.SGID:
           return validateSgidForm(form).andThen(() => {
+            const { codeChallenge, codeVerifier } = generatePkcePair()
+            res.cookie(
+              SGID_CODE_VERIFIER_COOKIE_NAME,
+              codeVerifier,
+              SgidService.getCookieSettings(),
+            )
             return SgidService.createRedirectUrl(
               formId,
               Boolean(isPersistentLogin),
               [],
+              codeChallenge,
               encodedQuery,
             )
           })
         case FormAuthType.SGID_MyInfo:
           return validateSgidForm(form).andThen(() => {
+            const { codeChallenge, codeVerifier } = generatePkcePair()
+            res.cookie(
+              SGID_CODE_VERIFIER_COOKIE_NAME,
+              codeVerifier,
+              SgidService.getCookieSettings(),
+            )
```

Comment:
nit: since now we always generate the PKCE pair before redirection, I'm wondering if we should just wrap this in `SgidService.createRedirectUrl`. Concretely, it would look something like 

```typescript
validateSgidForm(form)
.andThen(() => SgidService.createRedirectUrl(...))
.andThen(({ redirectUrl, cookie: { name, val, options } }) => {
  res.cookie(name, val, options)
  return redirectUrl
})
```

It doesn't really save us much code because we still have to handle passing values via the neverthrow monad. But the main benefit here is that the specifics of the protocol between FormSG and SGID is wrapped within the SGID service layer, rather than the public form handler having to know about the internals.